### PR TITLE
#CU-2mtq0ng #xcmp to #statemine #playbook

### PR DESCRIPTION
**Pull request by dzmitry-lahoda**
this is playbook to do to have bidirectional channel opened with statemine. all extrinsics calls are encoded into hex and cay be decoded.

### CODE REVIEW

File: integration-tests/local-integration-tests/awesome-xcmp.md [hunk 1]
```diff
@@ -1,25 +1,35 @@
...
+## Format of messages
 
 - https://medium.com/polkadot-network/xcm-part-ii-versioning-and-compatibility-b313fc257b83
 - https://medium.com/polkadot-network/xcm-part-iii-execution-and-error-management-ceb8155dd166
 - https://github.com/paritytech/xcm-format/blob/master/README.md
 - https://research.web3.foundation/en/latest/polkadot/XCMP/index.html
 - https://medium.com/polkadot-network/xcm-the-cross-consensus-message-format-3b77b1373392
 
-### XCM(P) desing
+## XCM(P) desing
```
Comment (vimukthi-git) [inline integration-tests/local-integration-tests/awesome-xcmp.md:32]:
```suggestion
## XCM(P) design
```

File: rfcs/0003-xcmp-picasso-statemine.md [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+
+## Preparation
+
+One should have `Identity` on Kusama to create `Proposal` on  https://parachains.polkassembly.io/ . 
+
+Suggested amount is 50 KSM total for all operations on `Balance`.
+
+Picasso chain also should have Balance, better 22 KSM.
+
+## Picasso Goverannce create request
```
Comment (vimukthi-git) [inline rfcs/0003-xcmp-picasso-statemine.md:21]:
```suggestion
## Picasso Governance create request
```

File: rfcs/0003-xcmp-picasso-statemine.md [hunk 3]
```diff
@@ -0,0 +1,252 @@
+
```
Comment (vimukthi-git) [inline rfcs/0003-xcmp-picasso-statemine.md:1]: Just curious do we generally have a design process that involves RFCs, or this is just for this proposal for XCM?

File: rfcs/0003-xcmp-picasso-statemine.md [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+# Overview
+
+This proposal sugget to open bidirectional HRMP channel between Picasso and Statemine. This will enable crosschain communication between Picasso and Statemine to enable various use cases including crosschain token transfer.
```
Comment (vimukthi-git) [inline rfcs/0003-xcmp-picasso-statemine.md:4]: Is there any particular roadmap that describes what use cases Picasso is trying to address with this channel? Or is just about token transfers of KSM to Picasso?

File: rfcs/0003-xcmp-picasso-statemine.md [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+## References
+
+
+- https://kusama.polkassembly.io/referendum/163
+
+- https://kusama.polkassembly.io/referendum/164
+
+- https://acala.discourse.group/t/open-hrmp-channel-between-karura-and-statemine/451
+
+- https://acala.discourse.group/t/open-hrmp-cross-chain-communication-between-bifrost-and-karura-parachain/316/7
```
Comment (vimukthi-git) [inline rfcs/0003-xcmp-picasso-statemine.md:252]: Very useful links :)

File: rfcs/0003-xcmp-picasso-statemine.md [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+2. Statemine accept channel request and send request to Picasso
+3. Accept channel form Statemine to Picasso
+
+## Preparation
+
+One should have `Identity` on Kusama to create `Proposal` on  https://parachains.polkassembly.io/ . 
+
+Suggested amount is 50 KSM total for all operations on `Balance`.
+
+Picasso chain also should have Balance, better 22 KSM.
```
Comment (vimukthi-git) [inline rfcs/0003-xcmp-picasso-statemine.md:19]: Is this for accepting the channel? Which account on Picasso?

File: rfcs/0003-xcmp-picasso-statemine.md [hunk 7]
```diff
@@ -0,0 +1,252 @@
...
+    }
+  ]
+}
+```
+
+Where [Kusama.utility.batchAll](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fkusama-rpc.polkadot.io#/extrinsics) is encoded as `0x1802083c01270800003c0027080000e803000000900100`
+
+## Accept request from Statemine
+
+Accept channel from state mine `0x3c01e8030000`
```
Comment (vimukthi-git) [inline rfcs/0003-xcmp-picasso-statemine.md:236]: Can this fail on Picasso side? if so what would happen?

**Review by vimukthi-git — APPROVED** [review]

File: rfcs/0003-xcmp-picasso-statemine.md (hunk 3 above)
Comment (dzmitry-lahoda) [reply rfcs/0003-xcmp-picasso-statemine.md:1]: Acala did proposal (see reference in the end), they have forum for that. So just mimic their approach, so use PRs to dicuss.

File: rfcs/0003-xcmp-picasso-statemine.md (hunk 6 above)
Comment (dzmitry-lahoda) [reply rfcs/0003-xcmp-picasso-statemine.md:19]: will clarify,  it is upper bound to have Identi pallet entry for referenda proposal and sending XCM to Kusama. Upper bound.

File: rfcs/0003-xcmp-picasso-statemine.md (hunk 7 above)
Comment (dzmitry-lahoda) [reply rfcs/0003-xcmp-picasso-statemine.md:236]: Yes. I guess this is out of scope of proposal how to debug issues. Can discuss in other place.

**Review by vimukthi-git — APPROVED** [review]

**Pull request merged**